test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/InvalidDownCallTests.java

```diff
@@ -174,4 +163,15 @@ public void test_heapSegmentForStructArgument() throws Throwable {
 			Assert.assertEquals(intHandle2.get(resultSegmt, 0L), 89113354);
 		}
 	}
+
+	@Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Heap segment not allowed.*")
+	public void test_disallowedHeapSegmentForPtrArgument() throws Throwable {
+		FunctionDescriptor fd = FunctionDescriptor.of(JAVA_INT, JAVA_INT, ADDRESS);
+		MemorySegment functionSymbol = nativeLibLookup.find("addIntAndIntsFromStructPointer").get();
+		MethodHandle mh = linker.downcallHandle(functionSymbol, fd, Linker.Option.critical(false));
+
+		MemorySegment structSegmt = MemorySegment.ofArray(new int[]{11121314, 15161718});
+		int result = (int)mh.invoke(19202122, structSegmt);
+		fail("Failed to throw out IllegalArgumentException when the heap segment is disallowed in native");
```

Comment:
nit: "out" could be removed to improve the failure message.

Reply:
Fixed everywhere.